docs/src/adoc/transaction.adoc

```diff
@@ -0,0 +1,52 @@
+= Transactions
+
+*jdbi* provides full support for JDBC transactions.
+
+== Transactions in the Fluent API
+
+*Handle* objects provide two ways to open a transaction -- *inTransaction* allows
+you to return a result, and *useTransaction* has no return value.
+
+Both optionally allow you to specify the transaction isolation level.
+
+[source,java,indent=0]
+-------------------------------------------
+include::TransactionTest.java[tags=simpleTransaction]
+-------------------------------------------
+
+Here, we (probably unnecessarily) guard a simple _SELECT_ statement with a
+transaction.
+
+== Transactions in SqlObject API
+
+Similarly, you may declare transactions with SqlObject annotations:
+
+[source,java,indent=0]
+-------------------------------------------
+include::TransactionTest.java[tags=sqlObjectTransaction]
+-------------------------------------------
```

Comment:
I do feel like an example where you do multiple operations imperatively in a `@Transaction` method (possibly calling other `@SqlUpdate` methods) would be more à propos for demonstrating this feature. But I'm fine with doing that work in a separate PR, as long as it's tracked.